ntbs-service-unit-tests/Services/NotificationServiceTest.cs

```diff
@@ -289,6 +289,22 @@ public async Task UpdateImmunosuppressionDetails_StripsOtherDescriptionWhenHasOt
             VerifyUpdateDatabaseCalled();
         }
 
+        [Fact]
+        public async Task UpdatePatientDetails_FormatsNhsNumber()
+        {
+            var reference = new PatientDetails { NhsNumber = "12345" };
+            var input = new PatientDetails { NhsNumber = "12 345" };
+            var notification = new Notification();
+
+            await _notificationService.UpdatePatientDetailsAsync(notification, input);
+
+            Assert.Equal(reference.NhsNumber, input.NhsNumber);
+
+            _mockContext.Verify(context => context.SetValues(notification.PatientDetails, input));
+            _mockAlertService.Verify(x => x.AutoDismissAlertAsync<DataQualityBirthCountryAlert>(It.IsAny<Notification>()), Times.Once);
+            VerifyUpdateDatabaseCalled();

```

Comment:
This test is a little bit limited in what it's testing - you're testing that the input object is correctly mutated, and that it is saved... but if those two things happened in the wrong order this test would still pass, while the wrong value would be saved.

I think we can do a bit better if we replace the verify call with:
```
_mockContext.Verify(context => context.SetValues(
    notification.PatientDetails,
    It.Is<PatientDetails>(pd => pd.NhsNumber == reference.NhsNumber)));
```

We could then get rid of the `Assert.Equal` call, as it isn't necessary. (And as an aside, I don't think the alert service verify adds anything to this test).

One final, very minor, remark is that `reference` here is probably better replaced by a simple string called `expectedNhsNumber`.

Reply:
Made these changes now. I assumed the `Verify(..)` before was checking that `input` was passed in with the formatted nhs number as I was passing that into `Verify(..)`, but even if so this is much clearer.